checker/src/org/checkerframework/checker/signedness/SignednessVisitor.java

```diff
@@ -91,6 +106,73 @@ private boolean maskIgnoresMSB(Kind maskKind, LiteralTree numBitsLit, LiteralTre
     }
 
     /**
+     * Given a casted right shift of the form {@code (type) (baseExpr >> numBits)} or {@code (type)
+     * (baseExpr >>> numBits)}, return true iff the expression's value is the same regardless of the
+     * type of right shift (signed or unsigned). This is true if the cast ignores the numBits most
+     * significant bits of the shift result -- that is, if the cast ignores all the new bits that
+     * the right shift introduced on the left.
+     *
+     * <p>For example, the function returns true for
+     *
+     * <pre>{@code (short) (myInt >> 16)}</pre>
+     *
+     * and for
+     *
+     * <pre>{@code (short) (myInt >>> 16)}</pre>
+     *
+     * because these two expressions are guaranteed to have the same result.
+     *
+     * @param shiftTypeKind the kind of the type of the shift literal (BYTE, CHAR, SHORT, INT, or
+     *     LONG)
+     * @param castTypeKind the kind of the cast target type (BYTE, CHAR, SHORT, INT, or LONG)
+     * @param numBitsLit the LiteralTree whose value is numBits
+     * @return true iff introduced bits are discarded
+     */
+    private boolean castIgnoresMSB(
+            TypeKind shiftTypeKind, TypeKind castTypeKind, LiteralTree numBitsLit) {
+        // Determine number of bits in the shift type, note shifts upcast to int
+        long shiftBits = 0;
+        switch (shiftTypeKind) {
+            case INT:
+                shiftBits = 32;
+                break;
+            case LONG:
+                shiftBits = 64;
+                break;
+            default:
+                return false;
+        }
+
+        // Determine number of bits in the cast type
+        long castBits = 0;
+        switch (castTypeKind) {
+            case BYTE:
+                castBits = 8;
+                break;
+            case CHAR:
+                castBits = 8;
+                break;
+            case SHORT:
+                castBits = 16;
+                break;
+            case INT:
+                castBits = 32;
+                break;
+            case LONG:
+                castBits = 64;
+                break;
+            default:
+                return false;
+        }
+
+        // Determine the bit difference between the shift and cast and the number of bits shifted
```

Comment:
This comment is not very helpful.  It just repeats what occurs below.  And, it is confusing because it describes two lines of code but the two occurrences of "and" are confusing to parse.